R/pkg/NAMESPACE

```diff
@@ -485,7 +485,6 @@ export("as.DataFrame",
        "cacheTable",
        "clearCache",
        "createDataFrame",
-       "createExternalTable",
```

Comment:
Hmm .. I wouldn't remove it for now. We have a bunch of deprecated API, and we don't have an explicit plan to how to handle them, and we're not even sure if we're actually going for 4.0.0.

Reply:
For example, if we apply the same argument, we should also remove `SQLContext`, etc., etc that would be largely a breaking change, which I believe can be controversial.

Reply:
Certainly, there may be an argument to not remove some methods. But these can only be removed at major versions; removing one does not imply all deprecated methods must be removed, either. What do you mean by 'going for 4.0.0' and how does this relate to SQLContext? there is indeed not one single plan for all deprecated methods, but what is the argument against this particular one?

Reply:
Oh I apologize that I forgot we already discussed that the next Spark version would likely be 4.0.0 - my bad.

My concern is mainly about "Considerations when breaking APIs" at https://spark.apache.org/versioning-policy.html. We discussed about removing some of deprecated API including `SQLContext` at Spark 3.0.0 but we did not remove in the end because it might be too breaking changes. At that time, we had to remove and revert some of those PRs (see also https://github.com/apache/spark/pull/22843 and https://github.com/apache/spark/pull/27815).

I personally would like to have a bit of discussion in the mailing list (probably after 3.5.0 release) before we remove some, and find out that some of them are too much breaking changes.

Reply:
Yeah I think it's fair to hold for a more joined up discussion of what should be removed